mne/utils/mixin.py

```diff
@@ -254,12 +254,13 @@ def _keys_to_idx(self, keys):
                 self._check_metadata(metadata=md)
                 try:
                     # Try metadata
-                    mask = self.metadata.eval(keys[0], engine='python').values
+                    vals = self.metadata.query(keys[0],
+                                               engine='python').index.values
```

Comment:
```suggestion
                    vals = self.metadata.reset_index().query(keys[0],
                                               engine='python').index.values
```

Reply:
Thanks! I didn't commit directly because of formatting (again the char limit is hitting me). I've used black-style formatting because we are going to switch anyway.